appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.kt

```diff
@@ -144,16 +150,22 @@ abstract class AppIntroBase : AppCompatActivity(), AppIntroViewPager.OnNextPageR
     }
 
     /**
-     * Request for a specific set of permission at a given slide.
-     * @param permissions A list of permission Strings
-     * @param slideNumber The slide position where to ask for permission
+     * Called by the user to associate permissions with slides.
+     *
+     * @param permissions - Array of permissions
+     * @param slideNumber - The slide at which permission is to be asked.
+     * @param required - Whether the user can change this slide without granting the permissions.
      */
-    protected fun askForPermissions(permissions: Array<String>, slideNumber: Int) {
+    @JvmOverloads
+    protected fun askForPermissions(permissions: Array<String>, slideNumber: Int, required: Boolean = true) {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             if (slideNumber <= 0) {
                 error("Invalid Slide Number: $slideNumber")
             } else {
-                permissionsArray.add(PermissionWrapper(permissions, slideNumber))
+                val permission = PermissionWrapper(permissions, slideNumber, required)
+                if (!permissionsMap.containsKey(slideNumber)) {
```

Comment:
Commenting only here but you have 5 occurrencies.

Please replace `permissionsMap.containsKey(slideNumber)` with `slideNumber !in permissionMap`